src/types/timezone.jl

```diff
@@ -30,6 +31,37 @@ function _reload_cache!(cache::AbstractDict, compiled_dir::AbstractString)
     return cache
 end
 
+const _TZ_CACHE_INITIALIZED = Threads.Atomic{Bool}(false)
+function initialize_tz_cache()
+    # Write out our compiled tzdata representations into a scratchspace
+   desired_version = TZData.tzdata_version()
+
+   _COMPILED_DIR[] = if desired_version == TZJData.TZDATA_VERSION
+       TZJData.ARTIFACT_DIR
+   else
+       # @info "Loading tzdata $desired_version"
```

Comment:
I commented out this because otherwise it just logs randomly when you call a function and the cache initializes. It feels a bit pointless in the first place as well.

Reply:
The point was to alert users that the environmental `JULIA_TZ_VERSION` was specified to a different version than TimeZones.jl would use by default. I'm not opposed to dropping this but it was informative

Reply:
Okay but it will just appear at some random point now without any context to the user.

Reply:
I think a reasonable compromise is to display a message during the `__init__` and perform the loading like you propose